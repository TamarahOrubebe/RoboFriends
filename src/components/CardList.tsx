import React from "react";
import Card from "./Card";

import { IRobot } from "./MainPage";

interface IProp {
  robots: Array<IRobot>;
}

const CardList = ({ robots }: IProp) => {
  return (
    <div>
      {robots.map((robot, i) => (
        <Card key={i} name={robot.name} email={robot.email} id={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
