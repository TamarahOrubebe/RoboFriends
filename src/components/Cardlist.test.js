import React from "react";
import { shallow } from "enzyme";
import CardList from "./CardList";


it("renders a CardList component", () => {

    const mockRobots = [
        {
            id: 1,
            username: 'Tamarah',
            name: 'Paul Orubebe',
            email: 'paulorubebe@gmail.com'
        }
    ]

    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
