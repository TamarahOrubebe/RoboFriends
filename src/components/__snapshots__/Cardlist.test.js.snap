// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CardList component 1`] = `
<div>
  <Card
    email="paulorubebe@gmail.com"
    id={1}
    key="0"
    name="Paul Orubebe"
  />
</div>
`;
